File: code/ch17/17.3.2.dayOfTheWeek.cpp
#include <string>
using namespace std;

class Solution {
public:
  string dayOfTheWeek(int day, int month, int year) {
    string list_of_days[] = {
      "Saturday", "Sunday", "Monday", "Tuesday",
      "Wednesday", "Thursday", "Friday"
    };
    if (month == 1 || month == 2) {
      year -= 1;
      month += 12;
    }
    int y = year % 100, c = year / 100;
    int weekday = (day + 13 * (month + 1) / 5 + y + \
                  y / 4 + c / 4 + 5 * c) % 7;
    return list_of_days[weekday];
  }
};

File: code/ch17/17.3.2.dayOfTheWeek.py
class Solution:
    def dayOfTheWeek(self, day: int, month: int, year: int) -> str:
        list_of_days = [
            "Saturday",
            "Sunday",
            "Monday",
            "Tuesday",
            "Wednesday",
            "Thursday",
            "Friday",
        ]
        if month == 1 or month == 2:
            year = year - 1
            month = month + 12
        y = year % 100
        c = year // 100
        weekday = int((day + 13 * (month + 1) // 5 + y + y // 4 + c // 4 + 5 * c) % 7)
        required_day = list_of_days[weekday]
        return required_day
